src/radical/pilot/agent/scheduler/continuous.py

```diff
@@ -422,8 +423,12 @@ def schedule_task(self, task):
             #   - if the previous use included this node
             # If a tag exists, continue to consider this node if the tag was
             # used for this node - else continue to the next node.
-            if tag is not None and tag in self._tag_history:
-                if node_uid not in self._tag_history[tag]:
+            if tag is not None:
+                if tag in self._tag_history:
+                    if node_uid not in self._tag_history[tag]:
+                        continue
+                # for a new tag check that nodes were not used for previous tags
+                elif node_uid in self._tagged_nodes:
```

Comment:
@andre-merzky along with these changes I would propose to have an extra attribute `exclusive` within `tags`: `td['tags'] = {'colocate': <tag>, 'exclusive': False}`

It would have default value as `True` (to use nodes assigned to a certain tag exclusively), and if it is ok to share nodes among different tags then to set it `False`

```python
                is_exclusive = td.get('tags', {}).get('exclusive', True)
                ...
                if ...
                elif is_exclusive and node_uid in self._tagged_nodes:
                    continue
```